cairo/ethereum/cancun/fork.cairo

```diff
@@ -195,3 +221,322 @@ func make_receipt{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr: Kec
+    alloc_locals;
+
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    // Validate transaction
+    let is_valid = validate_transaction(tx);
+    with_attr error_message("InvalidBlock") {
+        assert is_valid.value = TRUE;
+    }
+
+    // Get sender account
+    let state = env.value.state;
+    let sender = env.value.origin;
+    let sender_account = get_account{state=state}(sender);
+
+    // Get relevant transaction data
+    local tx_gas: Uint;
+    local tx_data: Bytes;
+    local tx_to: To;
+    local tx_value: U256;
+    local blob_gas_fee: Uint;
+    if (tx.value.blob_transaction.value != 0) {
+        assert tx_gas = tx.value.blob_transaction.value.gas;
+        assert tx_data = tx.value.blob_transaction.value.data;
+        assert tx_to = To(
+            new ToStruct(bytes0=cast(0, Bytes0*), address=&tx.value.blob_transaction.value.to)
+        );
+        assert tx_value = tx.value.blob_transaction.value.value;
+        let blob_gas_fee_res = calculate_data_fee(env.value.excess_blob_gas, tx);
+        assert blob_gas_fee = blob_gas_fee_res;
+        tempvar range_check_ptr = range_check_ptr;
+    } else {
+        tempvar range_check_ptr = range_check_ptr;
+    }
+
+    let range_check_ptr = [ap - 1];
+
+    if (tx.value.fee_market_transaction.value != 0) {
+        assert tx_gas = tx.value.fee_market_transaction.value.gas;
+        assert tx_data = tx.value.fee_market_transaction.value.data;
+        assert tx_to = tx.value.fee_market_transaction.value.to;
+        assert tx_value = tx.value.fee_market_transaction.value.value;
+        assert blob_gas_fee = Uint(0);
+    }
+
+    if (tx.value.legacy_transaction.value != 0) {
+        assert tx_gas = tx.value.legacy_transaction.value.gas;
+        assert tx_data = tx.value.legacy_transaction.value.data;
+        assert tx_to = tx.value.legacy_transaction.value.to;
+        assert tx_value = tx.value.legacy_transaction.value.value;
+        assert blob_gas_fee = Uint(0);
+    }
+
+    if (tx.value.access_list_transaction.value != 0) {
+        assert tx_gas = tx.value.access_list_transaction.value.gas;
+        assert tx_data = tx.value.access_list_transaction.value.data;
+        assert tx_to = tx.value.access_list_transaction.value.to;
+        assert tx_value = tx.value.access_list_transaction.value.value;
+        assert blob_gas_fee = Uint(0);
+    }
+
+    let effective_gas_fee = tx_gas.value * env.value.gas_price.value;
+
+    // Calculate available gas
+    let intrinsic_cost = calculate_intrinsic_cost(tx);
+    let gas = tx_gas.value - intrinsic_cost.value;
+
+    // Increment nonce
+    increment_nonce{state=state}(sender);
+
+    // Deduct gas fee from sender
+    with_attr error_message("OverflowError") {
+        assert sender_account.value.balance.value.high = 0;
+    }
+    let sender_balance_after_gas_fee = sender_account.value.balance.value.low - effective_gas_fee -
+        blob_gas_fee.value;
```

Comment:
do we know if this can be problematic to have accounts with very high balances?

Reply:
good question, eels assume they are uint

Reply:
Uint is actually un-bounded so I believe this can be problematic... we'll figure out